tests/bot/exts/info/test_tags.py

```diff
@@ -0,0 +1,430 @@
+    async def test_head_command(self):
+        """Should invoke `!tags get` command from `!tag` command."""
+        self.assertIsNone(await self.cog.tags_group.callback(self.cog, self.ctx, tag_name="class"))
+        self.ctx.invoke.assert_awaited_once_with(self.cog.get_command, tag_name="class")
+
+    async def test_search_tags_with_keyword_command(self):
+        """Should call `Tags._get_tags_via_content` and `Tags._send_matching_tags` with correct parameters."""
+        self.cog._get_tags_via_content = MagicMock(return_value="foo")
+        self.cog._send_matching_tags = AsyncMock()
+
+        self.assertIsNone(await self.cog.search_tag_content.callback(self.cog, self.ctx, keywords="youtube,audio"))
+        self.cog._get_tags_via_content.assert_called_once_with(all, "youtube,audio", self.member)
+        self.cog._send_matching_tags.assert_awaited_once_with(self.ctx, "youtube,audio", "foo")
+
+    async def test_search_tags_any_command(self):
+        """Should call `Tags._get_tags_via_content` and `Tags._send_matching_tags` with correct parameters."""
+        test_cases = [
+            {"keywords": "youtube,discord,foo"},
+            {"keywords": "any"}
+        ]
+        self.cog._get_tags_via_content = MagicMock(return_value="foo")
+        self.cog._send_matching_tags = AsyncMock()
+
+        for case in test_cases:
+            with self.subTest(keywords=case["keywords"]):
+                self.cog._get_tags_via_content.reset_mock()
+                self.cog._send_matching_tags.reset_mock()
+
+                self.assertIsNone(
+                    await self.cog.search_tag_content_any_keyword.callback(
+                        self.cog, self.ctx, keywords=case["keywords"]
+                    )
+                )
+                self.cog._get_tags_via_content.assert_called_once_with(any, case["keywords"] or "any", self.member)
+                self.cog._send_matching_tags.assert_awaited_once_with(self.ctx, case["keywords"], "foo")
+
+    async def test_send_matching_tags(self):
+        """Should return `None` and send correct embed."""
+        cache = self.cog._cache
+        test_cases = [
+            {
+                "args": (self.ctx, "youtube,audio", [cache["ytdl"]]),
+                "expected": Embed.from_dict(cache["ytdl"]["embed"])
+            },
+            {
+                "args": (self.ctx, "foo", []),
+                "expected": None
+            },
+            {
+                "args": (self.ctx, "bar", [cache["ytdl"], cache["class"], cache["classmethod"]]),
+                "expected": Embed.from_dict({
+                    "description": (
+                        "\n**»**   class\n"
+                        "**»**   classmethod\n"
+                        "**»**   ytdl\n"
+                    ),
+                    "footer": {"text": tags.FOOTER_TEXT},
+                    "title": "Here are the tags containing the given keyword:"
+                })
+            }
```

Comment:
I'm guessing `from_dict` is used because you cannot pass the footer text to the embed constructor. An alternative approach would be to build the embed before the list, though this is probably ok.

Reply:
`Embed(...).set_footer(text=tags.FOOTER_TEXT)` would work too since [`Embed.set_footer`](https://discordpy.readthedocs.io/en/latest/api.html#discord.Embed.set_footer) returns the instance.